src/resources/Catalogs/CatalogInterfaces.ts

```diff
@@ -14,6 +14,20 @@ export type HierarchyWithFields<T extends {fields?: string[]}> = Omit<T, 'fields
     fields: string[];
 };
 
+interface Cached<T> {
```

Comment:
I think it would be worth moving this interface in the [BaseInterface file](https://github.com/coveo/platform-client/blob/master/src/resources/BaseInterfaces.ts) if you think cached resources is something common across the services. If its only specific to the Commerce backend I would leave it there.

Reply:
It's currently only specific to a single route in the Commerce backend. For this reason, I'd rather keep it here for now.